subsys/modem/modem_ppp.c

```diff
@@ -192,40 +192,38 @@ static uint8_t modem_ppp_wrap_net_pkt_byte(struct modem_ppp *ppp)
 	return 0;
 }
 
+static bool modem_ppp_process_state_frame_byte(struct modem_ppp *ppp, uint8_t byte,
+					       uint8_t expected_byte)
+{
+	if (byte == expected_byte) {
+		ppp->receive_state++;
+		return true;
+	}
```

Comment:
I would prefer this approach:
```
static bool modem_ppp_process_is_byte_expected(uint8_t byte, uint8_t expected_byte)
        if (byte == expected_byte) {
	        return true;
        }

        LOG_DBG("Dropping byte 0x%2hhx because 0x%2hhx was expected.", byte, expected_byte);
	return false;
}
```
...
```
{
	switch (ppp->receive_state) {
	case MODEM_PPP_RECEIVE_STATE_HDR_SOF:
                if (modem_ppp_process_is_byte_expected(byte, MODEM_PPP_CODE_DELIMITER)) {
                        ppp->receive_state = MODEM_PPP_RECEIVE_STATE_HDR_FF;
                }
		break;
```
As this is easier to follow than hiding the new state within the helper, and using ++ instead of setting it directly. It will take up a few more bytes of ROM, but in this case the readability is more important IMO.